operator/pkg/ciliumendpointslice/manager.go

```diff
@@ -0,0 +1,635 @@
+// Copyright 2021 Authors of Cilium
```

Comment:
(Unrelated to this line of code)

@pchaigno raised a concern with this PR that the entire functionality is included in one very large commit, which we usually try to avoid for a few reasons: for instance smaller commits are easier to review; it allows us to document the reasoning behind individual changes over time; and if there's a regression, it's easier to locate the cause of the regression.

In this case, I think that the functionality overall is quite well self-contained, it was already reviewed in smaller chunks on the feature branch, and hopefully the way that it's relatively self-contained will minimize the potential blast radius. If there are any substantial bugs in the core implementation, I hope that you will work with the broader community to dig to the bottom of them (which I expect is in your interests as well, as the submitter of this functionality :-) ).

I raise this just to let you know that this is something reviewers generally will look for when reading PRs, so I'd encourage doing this in future wherever possible, even for a PR like this which merges a feature branch into the master branch. At the same time I don't think that forcing you to split up this particular PR now would make any substantial difference to the above parameters. Furthermore we never set this out explicitly as a requirement to begin with so it'd be perhaps a bit surprising to you if we required the changes to be split into smaller commits at this stage. Therefore I think I _won't_ ask you to make such changes to this PR now, but will encourage you to think about this for subsequent submissions.